salt/modules/keyboard.py

```diff
@@ -63,21 +63,31 @@ def set_sys(layout):
 
         salt '*' keyboard.set_sys dvorak
     """
+    result = None
     if salt.utils.path.which("localectl"):
-        __salt__["cmd.run"]("localectl set-keymap {}".format(layout))
+        if __salt__["cmd.retcode"]("localectl set-keymap {}".format(layout)) == 0:
+            result = True
+        else:
+            result = False
+    #TODO: Make behaviour dependent on outcome of command below
+    # Right now "result = True" just mimics the old behaviour, to stop things
+    # breaking, but it's not the right thing to do as it ignores the retcode.
     elif "RedHat" in __grains__["os_family"]:
         __salt__["file.sed"](
             "/etc/sysconfig/keyboard", "^LAYOUT=.*", "LAYOUT={}".format(layout)
         )
+        result = True
     elif "Debian" in __grains__["os_family"]:
         __salt__["file.sed"](
             "/etc/default/keyboard", "^XKBLAYOUT=.*", "XKBLAYOUT={}".format(layout)
         )
+        result = True
     elif "Gentoo" in __grains__["os_family"]:
         __salt__["file.sed"](
             "/etc/conf.d/keymaps", "^keymap=.*", "keymap={}".format(layout)
         )
-    return layout
+        result = True
+    return result
```

Comment:
Sorry I missed this in my previous review. We still want to ensure we return `layout` for both this function and the `get_x` function until Argon

Can you also add a comment in both functions:

```
# salt.utils.versions.warn_until(Argon, "The set_sys function will return True instead of layout in the Argon release)
```

```
# salt.utils.versions.warn_until(Argon, "The get_x function will return True instead of layout in the Argon release)
```

The reason being thats how we detect deprecations for a given release and we can then change the return to True instead of layout in that release.